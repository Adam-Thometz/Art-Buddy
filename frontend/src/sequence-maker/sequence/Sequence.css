.Sequence {
  display: flex;
  flex-direction: column;
}

.Sequence-display {
  display: grid;
  grid-template-columns: repeat(4, 375px);
  margin: 0 auto;
}

.Sequence-options {
  display: flex;
  justify-content: space-between;
}