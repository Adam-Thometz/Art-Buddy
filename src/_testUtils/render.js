import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
