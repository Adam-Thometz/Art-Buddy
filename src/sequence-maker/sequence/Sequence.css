.Sequence {
  display: grid;
  grid-template-columns: repeat(4, 23%);
  gap: 2%;
  margin: 0 auto;
  width: 90%;
}

.Sequence-block {
  display: inline-block;
  position: relative;
  border: 10px solid;
  border-radius: 25px;
  margin: 10px;
  height: 20vh;
  width: 85%;
  padding: 15px;
  font-size: 1.5rem;
}

.Sequence-block-img {
  margin: 0 auto;
  height: 20vh;
  width: 12vw;
  display: block;
  cursor: pointer;
}

.Sequence-block button {
  width: 15%;
  margin: 5px auto;
  font-size: 1rem;
  border-width: 2px;
  position: absolute;
  top: 5%;
  left: 5%;
}

.playing {
  box-shadow: 0 0 50px red;
}