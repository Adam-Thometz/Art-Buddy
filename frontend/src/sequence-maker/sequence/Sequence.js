import React from "react";

import { useSelector, useDispatch } from "react-redux";

import './Sequence.css';

import SequenceBlock from "./SequenceBlock";
import Button from "../../_components/button/Button";

import colors from "../../_components/button/colorOrder";
import { playSequence, resetSequence } from "../../_redux/actions/actions";

const Sequence = () => {
  const sequence = useSelector(state => state.sequenceMaker.sequence);
  const dispatch = useDispatch()

  const handlePlay = () => {
    dispatch(playSequence())
  }

  const handleReset = () => {
    dispatch(resetSequence());
  }
  
  return (
    <div className="Sequence">
      {sequence.some(block => block !== null) ? <p>Click on a picture to hear the sound!</p> : null}
      <div className="Sequence-display">
        {sequence.map(block => (
          <SequenceBlock block={block} />
        ))}
      </div>
      <div className="Sequence-options">
        <Button outlineColor={colors[0]} onClick={handlePlay}>PLAY</Button>
        <Button outlineColor={colors[3]} onClick={handleReset}>RESET</Button>
      </div>
    </div>
  );
};

export default Sequence;