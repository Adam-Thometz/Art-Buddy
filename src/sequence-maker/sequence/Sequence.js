import React from "react";

import { useSelector } from "react-redux";

import './Sequence.css';

import SequenceBlock from "./sequence-block/SequenceBlock";

import colors from "_helpers/_utils/colorOrder";

const Sequence = () => {
  const { sequence } = useSelector(state => state.sequenceMaker);

  return (
    <section className="Sequence">
      {sequence.map((block, i) => (
        <SequenceBlock key={i} id={i} block={block} borderColor={colors[i%4]} />
      ))}
    </section>
  );
};

export default Sequence;