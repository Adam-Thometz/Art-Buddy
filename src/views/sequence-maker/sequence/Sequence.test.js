import Sequence from "./Sequence";

import { render } from "testUtils/render";

describe("Sequence component", () => {
  it("renders without crashing", () => {
    render(<Sequence />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Sequence />);
    expect(asFragment()).toMatchSnapshot();
  });
});
