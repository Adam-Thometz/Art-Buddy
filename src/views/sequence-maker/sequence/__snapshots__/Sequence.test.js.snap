// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sequence component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="Sequence"
  >
    <figure
      class="Sequence-block"
      data-testid="block1"
      id="0"
      style="border-color: #89eb0a;"
    />
    <figure
      class="Sequence-block"
      data-testid="block2"
      id="1"
      style="border-color: #ffe81a;"
    />
    <figure
      class="Sequence-block"
      data-testid="block3"
      id="2"
      style="border-color: #fd4d3a;"
    />
    <figure
      class="Sequence-block"
      data-testid="block4"
      id="3"
      style="border-color: #4d4f5e;"
    />
  </section>
</DocumentFragment>
`;
